templates/quarkus-camel-api-consumer/skeleton/pom.xml

```diff
@@ -0,0 +1,256 @@
+                        </systemPropertyVariables>
+                    </configuration>
+                </plugin>
+
+                <plugin>
+                    <groupId>${quarkus.platform.group-id}</groupId>
+                    <artifactId>quarkus-maven-plugin</artifactId>
+                    <version>${quarkus.platform.version}</version>
+                </plugin>
+
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-failsafe-plugin</artifactId>
+                    <version>${maven-surefire-plugin.version}</version>
+                </plugin>
+
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-jar-plugin</artifactId>
+                    <version>${maven-jar-plugin.version}</version>
+                </plugin>
+            </plugins>
+        </pluginManagement>
+        <plugins>
+            <plugin>
+                <groupId>${quarkus.platform.group-id}</groupId>
+                <artifactId>quarkus-maven-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>build</id>
+                        <goals>
+                            <goal>generate-code</goal>
+                            <goal>build</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+
+            <plugin>
+                <groupId>net.revelc.code.formatter</groupId>
+                <artifactId>formatter-maven-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>format</id>
+                        <goals>
+                            <goal>format</goal>
+                        </goals>
+                        <phase>process-sources</phase>
+                    </execution>
+                </executions>
+            </plugin>
+
+            <plugin>
+                <groupId>net.revelc.code</groupId>
+                <artifactId>impsort-maven-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>sort-imports</id>
+                        <goals>
+                            <goal>sort</goal>
+                        </goals>
+                        <phase>process-sources</phase>
+                    </execution>
+                </executions>
+            </plugin>
+
+            <!--
+            This plugin connects with the Service Registry and download
+            the contracts (OpenApi) used in the project -->
+            <plugin>
+                <groupId>io.apicurio</groupId>
+                <artifactId>apicurio-registry-maven-plugin</artifactId>
+                <executions>
+                  <execution>
+                    <phase>generate-sources</phase>
+                    <goals>
+                      <goal>download</goal> 
```

Comment:
I am missing something here, given that in this service you are exposing an API, shouldn't it be registered? for example, something like:

```
<plugin>
                <groupId>io.apicurio</groupId>
                <artifactId>apicurio-registry-maven-plugin</artifactId>
                <version>${registry.version}</version>
                <executions>
                  <execution>
                    <phase>generate-sources</phase>
                    <goals>
                      <goal>register</goal> 
                    </goals>
                    <configuration>
                      <registryUrl>${{ values. service_registry_provider_url }}</registryUrl> 
                      <artifactType>OPENAPI</artifactType>
                      <artifacts>
                        <schema1>https://raw.githubusercontent.com/croway/${{values.application.split("/")[1]}}-${{ values.api_id }}-api/main/specification.yaml</schema1> 
                      </artifacts>
                    </configuration>
                  </execution>
                </executions>
              </plugin>
```

a client application may use the download goal to get the latest (deployed openapi spec)

I may be wrong here, but I think the process should be the following:
- create an api spec on its own git repo (via software template)
- Create a quarkus server like this one (via software template)
- During the build on tekton of the template, the opena api spec is registered on apicurio

this way, you can trust apicurio that the latest openapi spec registered is the one implemented by the application, and client apps can rely on that. wdyt @davgordo

Reply:
In my opinion "register" should happen in api-spec pipeline. In the consumer I only download the spec from the registry.